main.go

```diff
@@ -148,10 +181,13 @@ func main() {
 		os.Exit(1)
 	}
 
+	// This lease is not related to leader election. This is to report the status of the controller
+	// for observability reasons.
```

Comment:
What observability reasons?

I think this comment is important, it is a confusing point. I just want to make sure we're describing the "why" correctly for ourselves later. When I hear "obsesrvability" here I might think of the observability component in ACM. Is that what this is for? Or I thought it was something to do with addon framework?

Reply:
Thanks for calling me out on this. I added a better description.